src/ewatercycle/parameter_set.py

```diff
@@ -0,0 +1,61 @@
+from pathlib import Path
+from typing import Set
+
+from pydantic import BaseModel
+
+from ewatercycle.util import to_absolute_path
+
+
+class ParameterSet(BaseModel):
+    """Container object for parameter set options."""
+
+    name: str = ""
+    """Name of parameter set"""
+    directory: Path
+    """Location on disk where files of parameter set are stored.
+    If Path is relative then relative to CFG.parameterset_dir."""
+    config: Path
+    """Model configuration file which uses files from
+    :py:attr:`~directory`. If Path is relative then relative to
+    CFG.parameterset_dir."""
+    doi: str = "N/A"
+    """Persistent identifier of parameter set. For a example a DOI
+    for a Zenodo record."""
+    target_model: str = "generic"
+    """Name of model that parameter set can work with."""
+    supported_model_versions: Set[str] = set()
+    """Set of model versions that are
+    supported by this parameter set. If not set then parameter set will be
+    supported by all versions of model"""
+
+    class Config:
+        extra = "forbid"
+
+    def __str__(self):
+        """Nice formatting of parameter set."""
+        return "\n".join(
+            [
+                "Parameter set",
+                "-------------",
+            ]
+            + [f"{k}={v!s}" for k, v in self.__dict__.items()]
+        )
+
```

Comment:
Can we please remove this now in favour of the default pydantic printing? Especially their integration with Rich makes things look much better(https://docs.pydantic.dev/usage/rich/): 

```py
In [10]: from rich import print
In [11]: print(CFG)
Configuration(
    grdc_location=PosixPath('.'),
    container_engine='docker',
    apptainer_dir=PosixPath('.'),
    singularity_dir=None,
    output_dir=PosixPath('.'),
    parameterset_dir=PosixPath('.'),
    parameter_sets={},
    ewatercycle_config=None
)
```

Reply:
For an end user I like that they now see `/bla` instead of  `PosixPath('/bla')`.

If we do that then the Forcing and ParameterSet and models have different __str__ outputs.  I think they should look similar. We should do them all on none of them?

Reply:
Rather do all then. But I do agree that `PosixPath`s are not so nice.

Reply:
Made same in 990eb31

The notebooks in docs/ still use old format, but created https://github.com/eWaterCycle/ewatercycle/issues/344 to make it even nicer.